vispy/geometry/polygon.py

```diff
@@ -85,6 +85,20 @@ def edges(self):
 
         If no vertices or faces are specified, the function returns None.
         """
+        if self._edges is None:
+            npts = self._vertices.shape[0]
+            if np.any(self._vertices[0] != self._vertices[1]):
+                # start != end, so edges must wrap around to beginning.
+                edges = np.empty((npts, 2), dtype=np.uint32)
+                edges[:, 0] = np.arange(npts)
+                edges[:, 1] = edges[:, 0] + 1
+                edges[-1, 1] = 0
+            else:
+                # start == end; no wrapping required.
+                edges = np.empty((npts-1, 2), dtype=np.uint32)
+                edges[:, 0] = np.arange(npts)
+                edges[:, 1] = edges[:, 0] + 1
+            self._edges = edges
```

Comment:
Would it be better to move this chunk of code to the setter and then in init use `self.edges = edges` (instead of the private)?

Reply:
I might be missing something, but `self._edges` is never read directly, it is always `self.edges`. Additionally `self._edges` is never written to directly except for init. This block of code is just a validation step of the edges being applied to the PolygonData object so shouldn't that validation happen at set (one time) rather than at read (possibly many times).

Reply:
Ok it makes more sense now because I realize that they are just autogenerated from the vertices. However, when you set `vertices` don't you need to set `self._edges` to `None` so they are regenerated?